src/hooks/useFocusRef.ts

```diff
@@ -1,14 +1,13 @@
-import { useRef } from 'react';
+import { useCallback } from 'react';
 
-import { useLayoutEffect } from './useLayoutEffect';
-
-export function useFocusRef<T extends HTMLOrSVGElement>(isSelected: boolean) {
-  const ref = useRef<T>(null);
-
-  useLayoutEffect(() => {
-    if (!isSelected) return;
-    ref.current?.focus({ preventScroll: true });
-  }, [isSelected]);
+export function useFocusRef<T extends HTMLOrSVGElement>(isSelected: boolean, isFocused: boolean) {
+  const ref = useCallback(
+    (el: T | null) => {
+      if (el === null || !isFocused) return;
+      el.focus({ preventScroll: true });
```

Comment:
Do we not want to update the ref here after focusing?

Reply:
I will be updated in the [cell ref](https://github.com/adazzle/react-data-grid/pull/3187/files#diff-322374a1392f0a82977b2cfe82a7e08639d0eb211395073067129f39fd9be244R23). It does make me wonder about the timing. What if the flag is set to false before the ref in the formatter is initialized. I think children are mounted first I believe and the code seems to work.